### ISSUE TITLE
Unicode characters in connection string confuse pyodbc under Python2 on Windows

### ISSUE THREAD

**Issue by gordthompson** [mkleehammer/pyodbc#351]
The following code works fine under Python3 ...

```python
# -*- coding: utf-8 -*-
import pyodbc
import sys
print('Python version ' + sys.version.replace('\n', ' '))
print('pyodbc version ' + pyodbc.version)
conn_str = (
    u'DRIVER=ODBC Driver 11 for SQL Server;'
    u'SERVER=.\\SQLEXPRESS;'
    u'DATABASE=comptabilité;'
    u'Trusted_Connection=yes;'
)
cnxn = pyodbc.connect(conn_str, autocommit=True)
crsr = cnxn.cursor()
print(crsr.execute("SELECT txt FROM table1 WHERE id=1").fetchval())
```

... producing

```
Python version 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:54:40) [MSC v.1900 64 bit (AMD64)]
pyodbc version 4.0.21
Bonjour!
```

However, the exact same code fails under Python2 with

```
Python version 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
pyodbc version 4.0.21
Traceback (most recent call last):
  File "C:/Users/Gord/PycharmProjects/py2pyodbc_demo/main.py", line 14, in <module>
    print(crsr.execute("SELECT txt FROM table1 WHERE id=1").fetchval())
  File "C:\Users\Gord\PycharmProjects\py2pyodbc_demo\venv\lib\encodings\__init__.py", line 100, in search_function
    level=0)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 77: ordinal not in range(128)
```

Note that the traceback shows the `print(crsr.execute(...` line as the one that's failing, not the `pyodbc.connect` line.

**Comment by gordthompson** [comment]
I just tested the same code on Linux with unixODBC 2.3.1 and did not reproduce the error. Both Python2 and Python3 worked.

**Comment by v-chojas** [comment]
https://github.com/mkleehammer/pyodbc/pull/357

**Comment by mkleehammer** [comment]
I've merged #357, though I haven't tested this exactly.  @gordthompson, can you test with the master branch?  If it works, we need to figure out if you need a release for this immediately or if we should buffer up a few fixes.

**Comment by gordthompson** [comment]
Confirmed that the issue was still there with 4.0.23. After removing that and then doing

```
pip install git+https://github.com/mkleehammer/pyodbc@master
```

my test code succeeded with

```
Python version 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
pyodbc version 4.0.24b1
Bonjour!
```

No need for an expedited release on my account. I've pretty much completely switched over to Python3 now.
